discretize/tree_ext.pyx

```diff
@@ -1314,8 +1314,11 @@ cdef class _TreeMesh:
 
         return sp.csr_matrix((V, (I,J)), shape=(self.ntFz, self.nC))
 
+    @property
```

Comment:
@jcapriot: I added the property decorator and cache the cellGradxStencil (to prevent name mangling or double underscores, I appended `Mat` to the cached ones)

Reply:
If construction is fast, we can also simply keep the decorator and not worry about caching the matrix. We just need to be able to treat it as a property in the SimPEG regularization

Reply:
So this will work, only because when we use the Class it is as a parent to the actual class. This would not necessarily work on it's own in Cython because you cannot add members to the class on the fly. If you look at how I have to initialize all of the other possible cached operators you can see examples of it.

Reply:
ok, in this case, do you think it is better just to simply use the `@property` decorator and skip the caching?

Reply:
(I was looking at the `nodalGrad` for an example - but now I see that you create the cdef object on the `_TreeMesh` --> missed that on the first go)

Reply:
I think it's just fine to cache them. Especially the stacked version in TreeMesh.py, as the stack operation is not inexpensive.

I think it sounds like you see it, it gets added as a cdef object at the beginning of the class, then is initialized to None in the __init__ method. Just have to think of cython cdef classes as a static thing, (like a class defined in c/c++).